plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/PositionDeleteFilter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.delete;
+
+import io.trino.plugin.deltalake.DeltaLakeColumnHandle;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.util.List;
+
+import static io.trino.plugin.deltalake.DeltaLakeColumnHandle.ROW_ID_COLUMN_NAME;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.util.Objects.requireNonNull;
+
+public final class PositionDeleteFilter
+{
+    private final Roaring64NavigableMap deletedRows;
+
+    public PositionDeleteFilter(Roaring64NavigableMap deletedRows)
+    {
+        this.deletedRows = requireNonNull(deletedRows, "deletedRows is null");
+    }
+
+    public RowPredicate createPredicate(List<DeltaLakeColumnHandle> columns)
+    {
+        int filePositionChannel = rowPositionChannel(columns);
+        return (page, position) -> {
+            long filePosition = BIGINT.getLong(page.getBlock(filePositionChannel), position);
```

Comment:
`rowPositionChannel` finds column of name ROW_ID_COLUMN_NAME which typically has  `io.trino.plugin.deltalake.DeltaLakeColumnHandle#MERGE_ROW_ID_TYPE` type (a Row), not a bigint.